test/CMakeLists.txt

```diff
@@ -54,6 +56,17 @@ if(BUILD_TESTING)
             PASS_REGULAR_EXPRESSION "dgemm_\\|[ \t]+100000\\|"
     )
 
+    # Add test of config_file
+    add_test(NAME test_config_file COMMAND ${PROJECT_BINARY_DIR}/test/dgemm/test_dgemm)
+    set_tests_properties(test_config_file 
+        PROPERTIES 
+            ENVIRONMENT "PEAK_TARGET_CONFIG_ENV=PEAK_TARGET_CONFIG_PATH;PEAK_TARGET_CONFIG=BLAS;${PRELOAD_VAR}"
```

Comment:
You probably want to test two separate functionality. The first one is `PEAK_TARGET_CONFIG=BLAS` only and the second one is `PEAK_TARGET_CONFIG_ENV=PEAK_TARGET_CONFIG_PATH` only. This test here doesn't really test the second case.